app/scripts/modules/core/src/pipeline/config/stages/core/ExecutionBarLabel.tsx

```diff
@@ -30,6 +67,16 @@ export class ExecutionBarLabel extends React.Component<IExecutionBarLabelProps>
     }
     if (executionMarker) {
       const LabelComponent = stage.labelComponent;
+      if (LabelComponent !== ExecutionBarLabel && !this.state.hydrated) {
+        const loadingTooltip = (<Tooltip id={stage.id}><Spinner size="small"/></Tooltip>);
+        return (
+          <span onMouseEnter={this.hydrate}>
```

Comment:
does the overlay/tooltip component API include some kind of `onShow` callback or similar that this could hook into?